### PR TITLE
internal(clean-scripts): take in account more use cases and options

### PR DESCRIPTION
**Pull request by fbac**
# Description

Include more options:
- dry-run: to check images/disk/isos to be deleted
- now: to run clean-libvirt-pool

Now the logic takes in account more use cases to prevent images/isos/disk from being mistakenly deleted.

## Type of change

Please select the appropriate options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change is a documentation update

## Testing

- [x] I have performed my own tests
- [x] Binary has to be tested thoroughly by another colleague

## Checklist

- [x] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not be merged unless it passes CI or there is a comment/update saying what testing was passed.
- [x] PRs should not be merged unless positively reviewed.

### CODE REVIEW

**Review by iranzo — APPROVED** [review]
LGTM, needs testing

**Comment by alknopfler** [comment]
where are the tests passed????
Another question...which is the card in Jira for this tasks?

File: hack/clean/clean-libvirt-pool.sh [hunk 1]
```diff
@@ -10,10 +10,13 @@ resourcesUsed=""
 getUsedResources() {
     for vm in $(kcli list vm -o json | jq -r .[].name); do
         # preserve ISO/iso used by existing vm's
-        resourcesUsed=$(printf "%s|%s|${resourcesUsed}" "${vm}.ISO" "${vm}.iso")
+        resourcesUsed=$(printf "%s|${resourcesUsed}" "${vm}.ISO")
+
+        # some vm's have a boot-ID iso - depending on how they were created
+        resourcesUsed=$(printf "%s|${resourcesUsed}" "$(kcli show vm ${vm} | grep iso: | awk '{print $2}' | sed 's#\/var\/lib\/libvirt\/images\/##g')")
```
Comment (alknopfler) [inline hack/clean/clean-libvirt-pool.sh:16]: if you change the delimiter you don't need to escape it?

Please don't merge, and wait until other colleagues could review it

Comment (fbac) [reply hack/clean/clean-libvirt-pool.sh:16]: Another colleague already reviewed the code. Also escaping the characters while changing the delimiter does not harm the code.

Comment (alknopfler) [reply hack/clean/clean-libvirt-pool.sh:16]: exactly, it does not harm the code, but it's a style and quality of code matter. If we are able to improve something, please do it. (if you don't have enough time I could do it, but let's change it). Also, regarding the tests, please add if you have manual tests to the PR description in the test section, just to ensure everything merged in main has been tested (at least in any way)

Comment (iranzo) [reply hack/clean/clean-libvirt-pool.sh:16]: I disagree, the main reason for code styling is not about code "being pretty" which is highly subjective and not a valid engineering approach, but actually serving a purpose of legibility. Also. it's worth mentioning that styling is *never* enforced, not even in the kernel [1]. Regarding quality, it does have the same quality with or without escaped characters.

In this case the escaped characters cover the, far but possible, problem that a potentially buggy sed release wouldn't escape the characters, and the delimiters are different just for legibility reasons. Anyone familiarized with sed should be capable of reading the command without major issues.

[1] https://www.kernel.org/doc/html/v4.10/process/coding-style.html

**Pull request merged**